network/host/dht_test.go

```diff
@@ -1095,3 +1104,249 @@ func TestNewDHT(t *testing.T) {
 	dht.Disconnect()
 	<-done
 }
+
+func TestDHT_AuthenticationRequest(t *testing.T) {
+	count := 2
+	port := 8000
+	var dhts []*DHT
+	done := make(chan bool)
+	var ids []string
+
+	for i := 0; i < count; i++ {
+		id, _ := id.NewIDs(1)
+		ids = append(ids, id[0].String())
+		st, s, tp, r, _ := realDhtParams(id, "127.0.0.1:"+strconv.Itoa(port))
+		address, _ := node.NewAddress("127.0.0.1:" + strconv.Itoa(port-1))
+		bootstrapNode := node.NewNode(address)
+		dht, err := NewDHT(st, s, tp, r, &Options{
+			BootstrapNodes: []*node.Node{
+				bootstrapNode,
+			},
+		},
+			relay.NewProxy())
+		port++
+		dhts = append(dhts, dht)
+		assert.NoError(t, err)
+	}
+
+	for _, dht := range dhts {
+		ctx, _ := NewContextBuilder(dht).SetDefaultNode().Build()
+		assert.Equal(t, 0, dht.NumNodes(ctx))
+		go func(dht *DHT) {
+			err := dht.Listen()
+			assert.Equal(t, "closed", err.Error())
```

Comment:
use const **closedMessage** coz kcpTransport use "broken pipe" message for connection close

Reply:
fixed